pkg/controller/infinispan/resources/service/service_controller.go

```diff
@@ -416,6 +427,9 @@ func computeSiteService(ispn *ispnv1.Infinispan) *corev1.Service {
 				TargetPort: intstr.IntOrString{IntVal: consts.CrossSitePort},
 			},
 		}
+		if exposeConf.Port > 0 {
```

Comment:
shouldn't we do this for all `expose types`? `NodePort` and `ClusterIP` has the field `Port` too

Reply:
For `NodePort` it doesn't make sense as external communication is performing via `NodePort` on all hosts. For `ClusterIP` it make sense but can guess situation why customer need to change port for internal k8s cluster communication.

Reply:
how about the same cluster cross-site (i.e. different namespaces/cluster names)?
 isn't this `port` be used independently of the expose type?

Reply:
Yes, but for the same cluster service name is unique and port doesn't matter in this case as we are operating with service:port combination